packages/core/react/src/WalletProvider.tsx

```diff
@@ -30,6 +32,19 @@ function getUserAgent() {
     return _userAgent;
 }
 
+function getInferredClusterFromEndpoint(endpoint?: string): Cluster {
+    if (endpoint == null) {
+        return 'mainnet-beta';
+    }
+    if (/devnet/i.test(endpoint)) {
+        return 'devnet';
+    } else if (/testnet/i.test(endpoint)) {
+        return 'testnet';
```

Comment:
This is the dumbest possible thing we can do, but will work Most of the Time™. If you have a URL that both does not contain these words _and_ does not use mainnet, then you can supply a custom `SolanaMobileWalletAdapter` that makes the correct choice of `cluster` for your use case.

Reply:
What should happen if the endpoint the dapp has configured is localhost / 127.0.0.1?

Reply:
The MWA spec doesn't strictly define the values for `cluster` for this reason: https://solana-mobile.github.io/mobile-wallet-adapter/spec/spec.html#authorize

IMO, the right thing to do here would be to send the RPC URI in the `cluster` parameter. If the wallet supports interacting with arbitrary clusters, it should record that RPC URI and return a valid auth token. If it does not support arbitrary cluster, it can return an error instead.

Reply:
> IMO, the right thing to do here would be to send the RPC URI in the cluster parameter.

Hmm. It may be a security concern for a wallet to trust an RPC URL from a dapp. It can't tell for sure what cluster you are signing with that way, or the RPC can lie about its genesis hash or anything else if you tried.